[Feature] Overzicht van wedstrijden #15

MauricePonte opened the pull request:
Added matchesView/Model 

Added View en ViewModel for Matches
Added unit test for filtering

Refactor on MatchDefinition

Fixed main menu to work for now
Fixed unittest involving MatchDefinitions

Commit:
Added matchesView/Model

Added View en ViewModel for Matches
Added unit test for filtering

Refactor on MatchDefinition

Fixed main menu to work for now
Fixed unittest involving MatchDefinitions

rodeknopje commented on Dartin/Dartin.Application/ViewModels/MatchesViewModel.cs:
```diff
@@ -0,0 +1,97 @@
+
+        public void Add()
+        {
+            ScreenManager.GetInstance().SwitchViewModel(new MatchDefinitionViewModel(new MatchDefinition()));
```
moet dit niet `ScreenManager.GetInstance().SwitchViewModel(new MatchDefinitionViewModel(null));` zijn je checkt namelijk later of de original match null is maar nu is de original match nooit null, correct me if I'm wrong tho.

MauricePonte replied:
Ik heb die null check eruit gehaald, dit vindt ik mooier/duidelijker  :)

Jacsquare reviewed (approved):
prima, zie foto volgens mij is return niet nodig
![afbeelding](https://user-images.githubusercontent.com/43779942/137013510-54b295fd-408a-4aab-96bb-1fe279a0e951.png)

tr4wzified commented on Dartin/Dartin.Application/ViewModels/MainMenuViewModel.cs:
```diff
@@ -1,30 +1,32 @@
         public void OnExit()
         {
-            throw new System.NotImplementedException();
+            throw new NotImplementedException();
```
Als je de ViewName met => doet zou ik deze methodes ook met => doen (probeer ik ook aan te houden bij simpele one liners). Dit is niet de reden waarom ik block tho, zie tweede comment

tr4wzified commented on Dartin/Dartin.Application/ViewModels/MatchesViewModel.cs:
```diff
@@ -0,0 +1,97 @@
+                    Matches.Add(match);
+                }
+            }
+        }
```
Ik zou deze Clear() en foreach herschrijven, kan gewoon in 1 line met linq, en sowieso de case insensitive comparator gebruiken.

Matches = State.Instance.Matches.Where(match => match.Name.Contains(filter, StringComparison.OrdinalIgnoreCase);

Die OrdinalIgnoreCase werkt alleen in .NET Core 2.1+ btw Maurice, voordat je m bij CS probeert

Waarom initialiseert de FilterPlayers de lijst als die nog niet bestaat? Is dat het werk van een filter/search?

Commit:
Merge branch 'main' into Overview-of-matches-#15

Commit:
Refactor still in progress

Added basic navigation functionality in MainMenu, MatchesView and MatchDefinitionView
Added correct properties to matchDefinition

made necessary changes to other viewmodels

Commit:
Dikke merghe

Commit:
Overview of matches + General refactor

Added MarkUp to Converters for easy XML use
Added proper navigation to MainMenu, Matches OverView and Player OverView

Commit:
Merge

Commit:
Merge fix

rodeknopje commented on Dartin/Dartin.Application/ViewModels/MatchDefinitionViewModel.cs:
```diff
@@ -1,166 +1,124 @@
+            //else
+            //    OriginalMatch = CurrentObject;
 
-            ScreenManager.GetInstance().SwitchViewModel(new ScoreboardViewModel(CurrentObject));
```
Zou dit terug kunnen, nu kunnen we geen match meer starten

MauricePonte replied:
Met nieuwe commit aangepast

rodeknopje commented on Dartin/Dartin.Application/ViewModels/PlayersViewModel.cs:
```diff
@@ -174,13 +174,9 @@ public void Add()
             ModalButtonText = "Add";
 
-            FirstName = string.Empty;
-            LastName = string.Empty;
```
Zou je deze terug kunnen zetten, hierdoor is de modal nu prefilled met de first and lastname die je de vorige keer hebt ingevuld, dit valt overigs ook buiten de scope

MauricePonte replied:
Jaa veel buiten de scope, dit was toegevoegd omdat Player een constructor had. Maar deze is niet nodig. De prefill kan dan worden toegevoegd in de XML

rodeknopje replied:
Dit gaat om de Add() die heeft geen prefill nodig maar op het moment dat deze modal tevoorschijn komt moeten de first and lastname allebij empty zijn, dat wordt nu met twoway binding gedaan in de xaml

MauricePonte replied:
Jaa doordat de Player geen Constructor meer heeft zijn die gwn leeg dus klopt het weer

rodeknopje replied:
Deze FistName en LastName zijn bindings naar de xaml textboxjes, dit heeft niet zoveel met de player constructor te maken. Maak nu maar is een player aan dan zie je dat hij geen naam krijgt. Deze waardes moeten dus wel aan de nieuwe player worden geassigned. Of je dat nu met een constructor doet of op een andere manier maakt niet uit. 

als je deze twee regels weer terughaald en dan die `new Player { FirstName = FirstName, LastName = LastName}` doet in de feedback hieronder doet hij het gwn weer.

MauricePonte replied:
teruggezet naar oude situatie

Commit:
SaveAndStartGame + SaveGameAndExit aangepast

rodeknopje commented on Dartin/Dartin.Application/ViewModels/PlayersViewModel.cs:
```diff
@@ -174,13 +174,9 @@ public void Add()
             PlayerAction = () =>
             {
-                State.Instance.Players.Add(new Player(FirstName, LastName));
+                State.Instance.Players.Add(new Player());
```
Als je nu een speler aanmaakt dan heeft hij dus geen naam terwijl je wel een naam invult in de modal, blijkbaar was de constructor wel nodig 😄, maar gebruik hier anders  `new Player { FirstName = FirstName, LastName = LastName}` als je perse zonder constructor wilt doen

MauricePonte replied:
Jaa je past direct die dingen aan in die View en je Back button zou de player (die dan word aangemaakt) uit de state moeten verwijderen, dus niet nodig dacht ik :) Anders krijg je spelers met FirstName LastName, toch?

rodeknopje replied:
ik snap niet helemaal wat je bedoelt, ik wil als ik een speler aanmaak toch gwn de firstname en lastname kunnen invoeren

MauricePonte replied:
Dat kan ook ? Je past de First en Last in die pop up aan van dit object, via de properties, Daar heb je dan geen constructor voor nodig

rodeknopje replied:
Klopt indd, zou je dat ook willen doen dat de First en LastName wel worden gezet, want als je nu dus een speler maakt krijgt deze geen naam

MauricePonte replied:
Teruggezet naar oude situatie

rodeknopje commented on Dartin/Dartin.Application/ViewModels/MatchDefinitionViewModel.cs:
```diff
@@ -1,180 +1,146 @@
+            OriginalMatch.Players.Add(SelectedPlayerTwo);
+            OriginalMatch.LegsToWinSet = CurrentObject.LegsToWinSet;
+            OriginalMatch.SetsToWin = CurrentObject.SetsToWin;
+            OriginalMatch.LegsToWinSet = CurrentObject.SetsToWin;
```
Denk dat dit niet helemaal klopt :D

MauricePonte replied:
Jaa want in de constructor maak ik een nieuw object (CurrentObject) die vul ik via de viewModel. (Zodat ik niet de State aanpas) en daarna moet ik dat weer terugzetten naar het originele object, zodat de state wel aangepast word.

rodeknopje replied:
ik bedoel de laatste regel :)

MauricePonte replied:
oeps :)

rodeknopje reviewed (changes requested):
test

Commit:
pr fix :)

ork44 reviewed (changes requested):
Blijf binnen je scope

ork44 commented on Dartin/Dartin.UnitTests/PlayersViewModelTests.cs:
```diff
@@ -17,10 +17,26 @@ public class PlayersViewModelTests
         [Fact]
         public void FilterPlayers()
         {
-            var player1 = new Player("yob", "ama");
```
Waarom is dit veranderd?

MauricePonte replied:
Teruggezet

ork44 commented on Dartin/Dartin.UnitTests/ScoreboardViewModelTest.cs:
```diff
@@ -153,47 +154,65 @@ public void TestComparePlayerScoreWithScoreToWinLeg()
 
-            SubmitTossInputs(vm, "t20", "t20", "t20");
-            SubmitTossInputs(vm, "t20", "t20", "t20");
-            SubmitTossInputs(vm, "t20", "t20", "t20");
```
Waarom zijn de tests gecomment?

MauricePonte replied:
Zijn kapot helaas, moet met de maker er even naar kijken. Staat op mn lijstje

Commit:
Player constructor teruggezet, tests niet aangepast

ork44 reviewed (approved):
bg
